api/Service/UserService.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using api.Models;
+using Microsoft.EntityFrameworkCore;
+using YonderfulApi.Data;
+
+namespace api.Service
+{
+  public class UserService : IUserService
+  {
+
+    private readonly DataContext _context;
+    public UserService(DataContext context)
+    {
+        _context = context;
+    }
+    public async Task<bool> DeleteUser(int id)
+    {
+        var user = await _context.Users.FindAsync(id);
+        if (user == null) return false;
+
+        _context.Users.Remove(user);
+        return await _context.SaveChangesAsync() > 0;
+
+    }
+
+    public async Task<User> GetUser(int employeeId)
+    {
+        var user = await _context.Users.FindAsync(employeeId);
+        return user;
+    }
+
+    public async Task<IList<User>> GetUserList()
+    {
+        var employeeList = await _context.Users.ToListAsync();
+        return employeeList;
+    }
+
+    public async Task<User> PostUser(User user)
+    {
+        if (await UserExists(user.Email)) return null;
+
+        var newUser = new User
```

Comment:
Since there's already a User object received as a parameter, this new assignment looks redundant and should be removed.